subiquity/client/controllers/identity.py

```diff
@@ -48,3 +49,10 @@ def done(self, identity_data):
             "IdentityController.done next_screen user_spec=%s",
             identity_data)
         self.app.next_screen(self.endpoint.POST(identity_data))
+
+    async def _validate_username(self, username):
```

Comment:
I would prefer not to store the result of the request in this object.
When the user is typing the username, there will likely be multiple `_validate_username` coroutines running concurrently. So it would not be clear what actual username `self.username_validation` corresponds to at at given time.

Reply:
Here I struggled the most. The way I found was to store the latest async task in the Form, as part of its state. Since for sure a couple of coroutines will be spawn due user editting the text field, at each new change the previous task is cancelled. When the `username_validation_state` method is called, we return the result of the last task launched, or `UsernameValidation.OK` by default. Handling this task object so explicitly uglyfied the API test code a little bit, IHMO, but I think we made the point that the validation result is deterministically retrieved.

Reply:
Thanks! This looks much better IMO :)